data/reference/syntax.adoc

```diff
@@ -0,0 +1,277 @@
+= AsciiDoc Syntax
+:icons: font
+:stem:
+:toc: left
+:url-docs: https://asciidoctor.org/docs
+:url-gem: https://rubygems.org/gems/asciidoctor
+
+A brief reference of the most commonly used AsciiDoc syntax.
+See {url-docs} to find the full documentation and reference.
+
+== Paragraphs
+
+A normal paragraph.
+Line breaks do not get preserved.
+// line comments, starting with //, are skipped
+
+A blank line separates paragraphs.
+
+[%hardbreaks]
+This paragraph bears the `hardbreaks` option.
+Notice line breaks are now preserved.
+
+ An indented (literal) paragraph disables text formatting,
+ preserves spaces and line breaks, and is displayed in a
+ monospaced font.
+
+[sidebar#id.role]
+Add a style, ID, and/or role to give a paragraph special meaning, like this sidebar.
+
+NOTE: An admonition paragraph, like this note, grabs the reader's attention.
+
+TIP: You can convert this document using `asciidoctor` to see the published output.
+
+== Text Formatting
+:hardbreaks:
+
+.Constrained (applied at word boundaries)
+*strong importance* (aka bold)
+_stress emphasis_ (aka italic)
+`literal text` (aka monospaced)
++passthrough text+ (substitutions disabled)
+`+literal passthrough text+` (literal text with substitutions disabled)
+
+.Unconstrained (applied anywhere)
+**C**reate+**R**ead+**U**pdate+**D**elete
+fan__freakin__tastic
+``mono``culture
+
+.Replacements
+A long time ago in a galaxy far, far away...
+(C) 1976 Arty Artisan
+I believe I shall--no, actually I won't.
+
+.Macros
+pass:c,m[The EU icon:flag[\] is blue & contains ************ in yellow arranged in a circle.]
+// where c=specialchars, q=quotes, a=attributes, r=replacements, m=macros, p=post_replacements, etc.
```

Comment:
etc. ? I think we should be exhaustive

Reply:
In this case, I'm not convinced. This is not an exhaustive reference. It's just supposed to provide enough context either to spark your memory or pique your interest.

Reply:
@mojavelinux I meant there's a finite number of substitutions right ? In the user manual we only mention specialchars, quotes, attributes, replacements, macros and post replacements. Are there more ?

Reply:
Those are all the individual substitutions. There are also groups.

Reply:
I see that's probably why I was confused because the list appeared to be complete. Maybe we could state the fact that the list of individual substitutions is complete.

```
// where c=specialchars, q=quotes, a=attributes, r=replacements, m=macros and p=post_replacements; substitution groups can also be used
```

Reply:
As I was saying, I just don't think this is the right place for that. The only reason the legend is even there is to provide some hints as to what the letters mean. We say at the top to refer to the documentation for the full reference. This reference is supposed to be brief.

Reply:
(I'm trying hard to keep this from getting too cluttered).

Reply:
No worries I think it's fair enough :+1: